### ISSUE TITLE
problem with RI-CLPM

### ISSUE THREAD

**Issue by gantonides** [JeroenDMulder/RI-CLPM#4]
I am running the RI-CLPM syntax provided by Jeroen de Mulder in lavaan in R.
The model runs well using the standard CLPM.
R gives an error when running RI-CLPM:

 RICLPM8.fit <- lavaan(RICLPM8, 
+   data = dat, 
+   missing = 'ML', 
+   meanstructure = T, 
+   int.ov.free = T
+ )
Error in lavaan(RICLPM8, data = dat, missing = "ML", meanstructure = T,  : 
  lavaan ERROR: missing observed variables in dataset: rix

The "problematic syntax":
# Create between components (random intercepts) rix=~1*ma2015+1*ma2016+1*ma2017+1*ma2018+1*ma2019+1*ma2020+1*ma2021+1*ma2022
riy=~1*finhh2015+1*finhh2016+1*finhh2017+1*finhh2018+1*finhh2019+1*finhh2020+1*finhh2021+1*finhh2022

What is wrong here?

Thanks, Gerrit

**Comment by JeroenDMulder** [comment]
Hi Gerrit, 
I believe that you are missing the asterisks between the 1's and the variable names. These asterisks denote that you fix the factor loadings of the `rix` factor to 1. So the syntax should read: `rix =~ 1*ma2015 + 1*ma2016 + 1*ma2017 + 1*ma2018 + 1*ma2019 + 1*ma2020 + 1*ma2021 + 1*ma2022`

**Comment by gantonides** [comment]
Thank you, Jeroen.
I appreciate your response.

I know you would say this. When I copied the syntax to the website the asterisks disappeared, but they were there when the syntax did not work.

Then I copied your exact syntax and adapted it to my model, and it worked! Maybe R is sensitive to minor typing issues. Earlier, I discovered a difference in apostrophs. Still, I don’t know why my syntax did not work. Your syntax on the website proved to very very useful. Thank you!

Best wishes,

Gerrit


Gerrit Antonides
Emeritus Professor of Economics of Consumers and Households
Hollandseweg 1, kr. 0110
6706 KN  Wageningen
The Netherlands
Tel. +31.317.482415
Tel. +31.317.484049 (secr.)

Https://orcid.org/0000-0002-9983-1263
Honorary member of the International Association for Research in Economic Psychology (IAREP) at www.iarep.org
Honorary member of the Society for the Advancement of Behavioral Economics (SABE) at www.sabeconomics.org
Web site: https://www.wur.nl/en/Persons/prof.dr.-G-Gerrit-Antonides.htm?subpage=publications


Op 8 dec 2023 om 13:14 heeft Jeroen Mulder ***@***.***> het volgende geschreven:

﻿

Hi Gerrit,
I believe that you are missing the asterisks between the 1's and the variable names. These asterisks denote that you fix the factor loadings of the rix factor to 1. So the syntax should read: rix =~ 1*ma2015 + 1*ma2016 + 1*ma2017 + 1*ma2018 + 1*ma2019 + 1*ma2020 + 1*ma2021 + 1*ma2022
